### PR TITLE
[14.0] shopfloor_checkout: show full list of products after select_line

### PR DESCRIPTION
**Pull request by JuMiSanAr**
Before, in some cases, when scanning / selecting a line in the `select_line` screen, only the selected lines would be displayed in the `select_package` screen.

Now, all lines are displayed and the scanned lines appear as selected.

ref: rau-144

### CODE REVIEW

**Comment by JuMiSanAr** [comment]
@jbaudoux I've corrected this issue in the following cases:
- When scanning a lot.
- When scanning a package.
- When selecting a line manually.

I understand this behavior should be the same in all these cases, but if one of them should behave differently please let me know.

NOTE: Waiting for validation to add the tests.

**Comment by jbaudoux** [comment]
@JuMiSanAr Tested and works properly. Can you update the unit tests?

**Review by jbaudoux — APPROVED** [review]

**Comment by jbaudoux** [comment]
@simahawk Can you drop a review on this one?

File: shopfloor/services/checkout.py [hunk 1]
```diff
@@ -478,15 +478,18 @@ def _select_lines_from_package(self, picking, selection_lines, package, **kw):
                     ),
                 },
             )
-        self._select_lines(lines)
+        related_lines = selection_lines.filtered(
+            lambda l: l.package_id != package and not l.shopfloor_checkout_done
+        )
+        lines = self._select_lines(lines, related_lines=related_lines)
```
Comment (jbaudoux) [inline shopfloor/services/checkout.py]: Just for this one, we should not show something out of the currently scanned package

**Review by jbaudoux — CHANGES_REQUESTED** [review]

File: shopfloor/services/checkout.py (hunk 1 above)
Comment (JuMiSanAr) [reply shopfloor/services/checkout.py]: Updated.

Comment (simahawk) [reply shopfloor/services/checkout.py]: Maybe I miss a piece of ctx but I don't understand this change. If you are asking for a specific line that's the only line that you should select, all the others shouldn't be shown on the select package screen IMO :thinking:

**Review by jbaudoux — APPROVED** [review]

File: shopfloor/services/checkout.py (hunk 1 above)
Comment (jbaudoux) [reply shopfloor/services/checkout.py]: he dropped this change. When you scan a package, you see only the content of the package. Like it has always been

File: shopfloor_packing_info/tests/test_checkout_scan_line.py [hunk 2]
```diff
@@ -26,6 +26,10 @@ def test_scan_line_package_ok_packing_info_filled_info(self):
         self._fill_stock_for_moves(move2, in_package=True)
         picking.action_assign()
         move_line = move1.move_line_ids
+        related_lines = picking.move_line_ids - move_line
         self._test_scan_line_ok(
-            move_line.package_id.name, move_line, packing_info=packing_info_text
+            move_line.package_id.name,
+            move_line,
+            related_lines,
+            packing_info=packing_info_text,
```
Comment (jbaudoux) [inline shopfloor_packing_info/tests/test_checkout_scan_line.py]: @JuMiSanAr I think you should also revert this change as you reverted the change on the package scan.

You can also rebase and squash

File: shopfloor/services/checkout.py [hunk 3]
```diff
@@ -628,8 +630,12 @@ def _select_line_move_line(self, picking, selection_lines, move_line):
             return self._select_lines_from_package(
                 picking, selection_lines, move_line.package_id
             )
-        self._select_lines(move_line)
-        return self._response_for_select_package(picking, move_line)
+
+        related_lines = selection_lines.filtered(
+            lambda l: not l.package_id and l.product_id != move_line.product_id
```
Comment (simahawk) [inline shopfloor/services/checkout.py:645]: @jbaudoux the change is still here or... I'm drunk :smile:  :thinking:

Comment (JuMiSanAr) [reply shopfloor/services/checkout.py:645]: But we don't make it here when we're scanning a package, this is part of `_select_line_move_line`. In fact, this will take us to the `select_package` screen, right?

Comment (jbaudoux) [reply shopfloor/services/checkout.py:645]: see the condition just above, if the line is in a package, it jumps to _select_lines_from_package

So this change here is correct

Comment (simahawk) [reply shopfloor/services/checkout.py:645]: Ok, but why do you want to show lines that have nothing to deal w/ the current line?
When you select a specific line that has no package you should not list any other line IMO.
But again... I might miss the ctx or the final goal.

Comment (jbaudoux) [reply shopfloor/services/checkout.py:645]: How can you pack the products without seeing the lines? After selecting the first line, I need to see the other lines otherwise I cannot select more than one line to put in a pack
